src/vision/shiny_future/barbara_makes_a_wish.md

```diff
@@ -0,0 +1,166 @@
+
+```
+% rustup install wish4-consolation
+...
+% consolation --port 8080
+```
+
+This brings up a terminal window that looks similar to the Unix `top` program, except that instead of a list of OS processes, this offers a list of tasks, and shows their status (i.e. running, ready to poll, or blocked), as well as some metrics about how long the tasks spend in each state.
+
+At the top of the screen, Barbara sees highlighted warning: "deadlock cycle was detected. hit `P` for more info."
+
+Barbara types capital `P`. The terminal switches to "problem view," which shows
+
+ * The task types, ID, and attributes for each type.
+ * The resources being awaited on
+ * The location / backtrace of the await.
+ * A link to a documentation page expanding on the issue.
+
+The screen also says "hit `D` to generate a graphviz `.dot` file to disk describing the cycle."
+
+Barbara hits `D` and stares at the resulting graph.
+
+Barbara suddenly realizes her mistake: She had constructed a single task that was sometimes enqueuing work (by sending messages on the channel), and sometimes dequeuing work, but she had not put any controls into place to ensure that the dequeuing (via `recv`) would get prioritized as the channel filled up.
+
+Barbara reflects on the matter: she knows that she could swap in an unbounded channel to resolve this, but she thinks that she would be better off thinking a bit more about her system design, to see if she can figure out a way to supply back-pressure so that the send rate will go down as the channel fills up.
+
+
+## 🤔 Frequently Asked Questions
+
+### **What status quo story or stories are you retelling?**
+
+[Barbara wants Async Insights](../status_quo/barbara_wants_async_insights.md)
+
+### **What is [Alan] most excited about in this future? Is he disappointed by anything?**
+
+Alan is happy to see a tool that gives one a view into the internals of the async executor.
+
+Alan is not so thrilled about using the `consolation` terminal interface; but luckily there are other options, namely a web-browser based client that offers even richer functionality, such as renderings of the task/resource dependency graph.
+
+### **What is [Grace] most excited about in this future? Is she disappointed by anything?**
+
+Grace is happy to see a tool, but wonders whether it could have been integrated into `gdb`.
+
+Grace is not so thrilled to learn that this tool is not going to try to provide specific insight into performance issues that arise solely from computational overheads in her own code. (The readme for `wish4-async-insight` says on this matter "for that, use perf," which Grace finds unsatisfying.)
+
+### **What is [Niklaus] most excited about in this future? Is he disappointed by anything?**
+
+Niklaus is happy to learn that the `wish4-async-insight` is supported by both `async-std` and `tokio`, since he relies on friends in both communities to help him learn more about Async Rust.
```

Comment:
This might not be possible to answer, but do we have a feeling for how likely it would be for alternative runtimes (e.g., specialized, bespoke runtimes) to integrate with this tool?

Reply:
Fixed in 671e42dfefd7832ece984b8b8e71e7fc7037cb38